Makefile

```diff
@@ -256,6 +256,27 @@ coverage:
 	echo -e $(BUILD_MSG) "build/$@" && \
 		$(ENV_SCRIPT) ./scripts/run_cov.sh -y
 
+################
+## Formatting ##
+################
+.PHONY: check_nph_installed check_precommit_hook install_nph_prehook
+
+check_nph_installed:
+ifeq (, $(shell which nph 2> /dev/null))
```

Comment:
This is nice indeed! Nevertheless, I think it is better to have this check implicit when the developer performs any `make` command. I mean, this is a pre-condition that should always be met and I think it will be more secure to enforce that in a generic manner.

Reply:
I would disagree with that. If I am a user that just want to build `wakunode2` because I want latest master or testing a branch, the presence of `nph` should not block me to proceed.